File: ClassEval_T/java/solutuon/Thermostat.java
package org.example;

import java.util.Objects;

public class Thermostat {
    double currentTemperature;
    private double targetTemperature;
    private String mode;

    public Thermostat(double currentTemperature, double targetTemperature, String mode) {
        this.currentTemperature = currentTemperature;
        this.targetTemperature = targetTemperature;
        this.mode = mode;
    }

    public double getTargetTemperature() {
        return targetTemperature;
    }

    public void setTargetTemperature(double temperature) {
        this.targetTemperature = temperature;
    }

    public String getMode() {
        return mode;
    }

    public boolean setMode(String mode) {
        if (Objects.equals(mode, "heat") || Objects.equals(mode, "cool")) {
            this.mode = mode;
            return true;
        } else {
            return false;
        }
    }

    public void autoSetMode() {
        if (currentTemperature < targetTemperature) {
            this.mode = "heat";
        } else {
            this.mode = "cool";
        }
    }

    public boolean autoCheckConflict() {
        if (currentTemperature > targetTemperature) {
            if (Objects.equals(mode, "cool")) {
                return true;
            } else {
                autoSetMode();
                return false;
            }
        } else {
            if (Objects.equals(mode, "heat")) {
                return true;
            } else {
                autoSetMode();
                return false;
            }
        }
    }

    public int simulateOperation() {
        autoSetMode();
        int useTime = 0;
        if (Objects.equals(mode, "heat")) {
            while (currentTemperature < targetTemperature) {
                currentTemperature += 1;
                useTime += 1;
            }
        } else {
            while (currentTemperature > targetTemperature) {
                currentTemperature -= 1;
                useTime += 1;
            }
        }
        return useTime;
    }
}

File: ClassEval_T/cpp/solution/Thermostat.cpp
#include <string>

class Thermostat {
public:
    Thermostat(double current_temperature, double target_temperature, const std::string& mode);

    double get_target_temperature() const;
    void set_target_temperature(double temperature);

    std::string get_mode() const;
    bool set_mode(const std::string& mode);

    void auto_set_mode();
    bool auto_check_conflict();
    int simulate_operation();

    double current_temperature;
    double target_temperature;
    std::string mode;
};
Thermostat::Thermostat(double current_temperature, double target_temperature, const std::string& mode)
    : current_temperature(current_temperature), target_temperature(target_temperature), mode(mode) {}

double Thermostat::get_target_temperature() const {
    return target_temperature;
}

void Thermostat::set_target_temperature(double temperature) {
    target_temperature = temperature;
}

std::string Thermostat::get_mode() const {
    return mode;
}

bool Thermostat::set_mode(const std::string& new_mode) {
    if (new_mode == "heat" || new_mode == "cool") {
        mode = new_mode;
        return true;
    }
    return false;
}

void Thermostat::auto_set_mode() {
    if (current_temperature < target_temperature) {
        mode = "heat";
    }
    else {
        mode = "cool";
    }
}

bool Thermostat::auto_check_conflict() {
    if (current_temperature > target_temperature) {
        if (mode == "cool") {
            return true;
        }
        else {
            auto_set_mode();
            return false;
        }
    }
    else {
        if (mode == "heat") {
            return true;
        }
        else {
            auto_set_mode();
            return false;
        }
    }
}

int Thermostat::simulate_operation() {
    auto_set_mode();
    int use_time = 0;
    if (mode == "heat") {
        while (current_temperature < target_temperature) {
            current_temperature += 1;
            use_time += 1;
        }
    }
    else {
        while (current_temperature > target_temperature) {
            current_temperature -= 1;
            use_time += 1;
        }
    }
    return use_time;
}
